packages/zero-pg/src/schema.ts

```diff
@@ -0,0 +1,82 @@
+import type {ServerSchema} from '../../z2s/src/schema.ts';
+import {formatPg, sql} from '../../z2s/src/sql.ts';
+import type {Schema} from '../../zero-schema/src/builder/schema-builder.ts';
+import type {DBTransaction} from '../../zql/src/mutate/custom.ts';
+
+export type ServerSchemaRow = {
+  schema: string;
+  table: string;
+  column: string;
+  type: string;
+  enum: string;
+};
+
+export async function getServerSchema<S extends Schema>(
+  dbTransaction: DBTransaction<unknown>,
+  schema: S,
+): Promise<ServerSchema> {
+  const serverToClient: Record<string, string> = {};
+  const schemaTablePairs: [string, string][] = [];
+  for (const {name, serverName} of Object.values(schema.tables)) {
+    let schemaTablePair: [string, string] = ['public', serverName ?? name];
+    if (serverName) {
+      const firstPeriod = serverName.indexOf('.');
+      if (firstPeriod > -1) {
+        schemaTablePair = [
+          serverName.substring(0, firstPeriod),
+          serverName.substring(firstPeriod + 1, serverName.length),
+        ];
+      }
+    }
+    schemaTablePairs.push(schemaTablePair);
+    serverToClient[`${schemaTablePair[0]}.${schemaTablePair[1]}`] = name;
+  }
+
+  if (schemaTablePairs.length === 0) {
+    return {}; // No pairs to query for
+  }
+  const inClause = sql.join(
+    schemaTablePairs.map(
+      ([schema, table]) => sql`(${schema}::text, ${table})::text`,
```

Comment:
Missing a `::text` cast?

`(${schema}::text, ${table}::text)::text`

Reply:
ah the one ::text should be in the parens